modules/plugins/snippets/luasnip/config.nix

```
@@ -9,12 +9,12 @@
 in {
   config = mkIf cfg.enable {
     vim = {
-      lazy.plugins = {
-        luasnip = {
-          package = "luasnip";
-          lazy = true;
-          after = cfg.loaders;
-        };
+      lazy.plugins.luasnip = {
+        package = "luasnip";
+        event = "BufEnter";
```

```suggestion
        lazy = true;
```
The reason we use `lazy = true` is because LuaSnip is automatically loaded when cmp_luasnip requires it and I don't see a reason to change that since using LuaSnip through nvim-cmp is the only supported way built into nvf.

---

> The reason we use `lazy = true` is because LuaSnip is automatically loaded when cmp_luasnip requires it, and I don't see a reason to change that since using LuaSnip through nvim-cmp is the only supported way built into nvf.

This causes a problem for people who do not use nvim-cmp though. It is not a hard dependency of luasnip, and it should be possible to use luasnip without completion, or with another plugin, such as blink, which i use.

---

Blink support is being implemented internally, but if you don't use blink it is still possible to:
- use the cmp_luasnip provider with blink
- modify the LuaSnip lazy definition in your configuration (this is the reason it's an attribute set)

---

> Blink support is being implemented internally, but if you don't use cmp it is still possible to:
> 
>     * use the cmp_luasnip provider with blink
> 
>     * modify the LuaSnip lazy definition in your configuration (this is the reason it's an attribute set)

Seeing as LuaSnip doesn't provide the cmp source itself, but requires the cmp_luasnip plugin, i don't think they shouldn't be connected here by default either, especially considering blink has its own source for luasnip built in, which does not trigger the lazy loading.

> Expand LSP-Snippets with [nvim-compe](https://github.com/hrsh7th/nvim-compe) (or its' successor, [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) (requires [cmp_luasnip](https://github.com/saadparwaiz1/cmp_luasnip)))

---

Why not? How does it use luasnip then?

Also, another option is to trigger the load after you load blink with `require("lz.n").trigger_load("luasnip")`

---

luasnip should just work with blink, remember to set `snippets.preset = 'luasnip'` in blink's config

---

> luasnip should just work with blink, remember to set `snippets.preset = 'luasnip'` in blink's config

That works as it should, it just doesn't trigger the lazy loading, i don't know the reason for that, but some other event (in my case BufEnter) had to trigger lazy loading for it to work